### ISSUE TITLE
no CUDA-capable device is detected

### ISSUE THREAD

**Issue by arnar-th** [immich-app/immich#10334]
### The bug

Smart search stops working couple of days (2-3) after I restart my immich stack.  Logs show 'no CUDA-capable device is detected' with extended errors (see included logs).

Running '> docker exec -it immich_machine_learning nvidia-smi' returns 'Failed to initialize NVML: Unknown Error'.
(Works as expected right after restart)

At the same time nvidia-smi shows everything working as expected on host and in other containers, jellyfin for example.

Taking down the stack and starting it again 'fixes' the issue for few days.

Ubuntu 22.04
Quadro T2000
Driver Version: 535.171.04
CUDA Version: 12.2

Do I need more memory (4gb) for the ViT-B-32__openai model?  Smart search does work after restart though so using the module should be ok.

This issue is not limited to v1.105.1, this happened in all previous versions since I enabled hardware acceleration (at least six months but I don't install every version).
I have removed all nvidia / cuda and reinstalled everything.  Also the fact that hardware acceleration still works in other containers makes me think this is immich related, possibly related to the model selected.

Possible duplicate of https://github.com/immich-app/immich/issues/9911

### The OS that Immich Server is running on

Ubuntu 22.04

### Version of Immich Server

v1.105.1

### Version of Immich Mobile App

v1.105.1

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
Relevant part from docker compose:
.
.
  immich-machine-learning:
    container_name: immich_machine_learning
    # For hardware acceleration, add one of -[armnn, cuda, openvino] to the image tag.
    # Example tag: ${IMMICH_VERSION:-release}-cuda
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}-cuda
    extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/ml-hardware-acceleration
      file: hwaccel.ml.yml
      service: cuda # set to one of [armnn, cuda, openvino, openvino-wsl] for accelerated inference - use the `-wsl` version for WSL2 where applicable
    volumes:
      - ./immich/model-cache:/cache
    env_file:
      - .env
    environment:
      - PUID=${PUID}
      - PGID=${PGID}
    restart: unless-stopped
.
.

Relevant part from hwaccel.ml.yml
.
.  cuda:
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities:
                - gpu
.
.
```

### Your .env content

```Shell
Relevant part:
.
.
IMMICH_VERSION=release
.
.
```

### Reproduction steps

```bash
1. Start immich docker stack
2. Wait few days
3. Try smart search
...
```

### Relevant log output

```shell
immich_machine_learning  | [06/12/24 16:17:31] INFO     Application startup complete.                      
immich_machine_learning  | [06/14/24 17:37:44] INFO     Setting 'ViT-B-32__openai' execution providers to  
immich_machine_learning  |                              ['CUDAExecutionProvider', 'CPUExecutionProvider'], 
immich_machine_learning  |                              in descending order of preference   
immich_machine_learning  | [06/14/24 17:37:44] INFO     Loading clip model 'ViT-B-32__openai' to memory    
immich_machine_learning  | *************** EP Error ***************
immich_machine_learning  | EP Error /onnxruntime_src/onnxruntime/core/providers/cuda/cuda_call.cc:121 std::conditional_t<THRW, void, onnxruntime::common::Status> onnxruntime::CudaCall(ERRTYPE, const char*, const char*, ERRTYPE, const char*, const char*, int) [with ERRTYPE = cudaError; bool THRW = true; std::conditional_t<THRW, void, onnxruntime::common::Status> = void] /onnxruntime_src/onnxruntime/core/providers/cuda/cuda_call.cc:114 std::conditional_t<THRW, void, onnxruntime::common::Status> onnxruntime::CudaCall(ERRTYPE, const char*, const char*, ERRTYPE, const char*, const char*, int) [with ERRTYPE = cudaError; bool THRW = true; std::conditional_t<THRW, void, onnxruntime::common::Status> = void] CUDA failure 100: no CUDA-capable device is detected ; GPU=28834 ; hostname=1736b8c4863e ; file=/onnxruntime_src/onnxruntime/core/providers/cuda/cuda_execution_provider.cc ; line=245 ; expr=cudaSetDevice(info_.device_id); 
immich_machine_learning  | 
immich_machine_learning  |  when using ['CUDAExecutionProvider', 'CPUExecutionProvider']
immich_machine_learning  | Falling back to ['CUDAExecutionProvider', 'CPUExecutionProvider'] and retrying.
immich_machine_learning  | ****************************************
immich_machine_learning  | [06/14/24 17:37:44] ERROR    Exception in ASGI application                      
immich_machine_learning  |                                                                                 
immich_machine_learning  |                              ╭─────── Traceback (most recent call last) ───────╮
immich_machine_learning  |                              │ /opt/venv/lib/python3.11/site-packages/onnxrunt │
immich_machine_learning  |                              │ ime/capi/onnxruntime_inference_collection.py:41 │
immich_machine_learning  |                              │ 9 in __init__                                   │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │    416 │   │   disabled_optimizers = kwargs["di │
immich_machine_learning  |                              │        kwargs else None                         │
immich_machine_learning  |                              │    417 │   │                                    │
immich_machine_learning  |                              │    418 │   │   try:                             │
immich_machine_learning  |                              │ ❱  419 │   │   │   self._create_inference_sessi │
immich_machine_learning  |                              │        disabled_optimizers)                     │
immich_machine_learning  |                              │    420 │   │   except (ValueError, RuntimeError │
immich_machine_learning  |                              │    421 │   │   │   if self._enable_fallback:    │
immich_machine_learning  |                              │    422 │   │   │   │   try:                     │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │ /opt/venv/lib/python3.11/site-packages/onnxrunt │
immich_machine_learning  |                              │ ime/capi/onnxruntime_inference_collection.py:48 │
immich_machine_learning  |                              │ 3 in _create_inference_session                  │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │    480 │   │   │   disabled_optimizers = set(di │
immich_machine_learning  |                              │    481 │   │                                    │
immich_machine_learning  |                              │    482 │   │   # initialize the C++ InferenceSe │
immich_machine_learning  |                              │ ❱  483 │   │   sess.initialize_session(provider │
immich_machine_learning  |                              │    484 │   │                                    │
immich_machine_learning  |                              │    485 │   │   self._sess = sess                │
immich_machine_learning  |                              │    486 │   │   self._sess_options = self._sess. │
immich_machine_learning  |                              ╰─────────────────────────────────────────────────╯
immich_machine_learning  |                              RuntimeError:                                      
immich_machine_learning  |                              /onnxruntime_src/onnxruntime/core/providers/cuda/cu
immich_machine_learning  |                              da_call.cc:121 std::conditional_t<THRW, void,      
immich_machine_learning  |                              onnxruntime::common::Status>                       
immich_machine_learning  |                              onnxruntime::CudaCall(ERRTYPE, const char*, const  
immich_machine_learning  |                              char*, ERRTYPE, const char*, const char*, int)     
immich_machine_learning  |                              [with ERRTYPE = cudaError; bool THRW = true;       
immich_machine_learning  |                              std::conditional_t<THRW, void,                     
immich_machine_learning  |                              onnxruntime::common::Status> = void]               
immich_machine_learning  |                              /onnxruntime_src/onnxruntime/core/providers/cuda/cu
immich_machine_learning  |                              da_call.cc:114 std::conditional_t<THRW, void,      
immich_machine_learning  |                              onnxruntime::common::Status>                       
immich_machine_learning  |                              onnxruntime::CudaCall(ERRTYPE, const char*, const  
immich_machine_learning  |                              char*, ERRTYPE, const char*, const char*, int)     
immich_machine_learning  |                              [with ERRTYPE = cudaError; bool THRW = true;       
immich_machine_learning  |                              std::conditional_t<THRW, void,                     
immich_machine_learning  |                              onnxruntime::common::Status> = void] CUDA failure  
immich_machine_learning  |                              100: no CUDA-capable device is detected ; GPU=28834
immich_machine_learning  |                              ; hostname=1736b8c4863e ;                          
immich_machine_learning  |                              file=/onnxruntime_src/onnxruntime/core/providers/cu
immich_machine_learning  |                              da/cuda_execution_provider.cc ; line=245 ;         
immich_machine_learning  |                              expr=cudaSetDevice(info_.device_id);               
immich_machine_learning  |                                                                                 
immich_machine_learning  |                                                                                 
immich_machine_learning  |                                                                                 
immich_machine_learning  |                              The above exception was the direct cause of the    
immich_machine_learning  |                              following exception:                               
immich_machine_learning  |                                                                                 
immich_machine_learning  |                              ╭─────── Traceback (most recent call last) ───────╮
immich_machine_learning  |                              │ /usr/src/app/main.py:116 in predict             │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │   113 │   except orjson.JSONDecodeError:        │
immich_machine_learning  |                              │   114 │   │   raise HTTPException(400, f"Invali │
immich_machine_learning  |                              │   115 │                                         │
immich_machine_learning  |                              │ ❱ 116 │   model = await load(await model_cache. │
immich_machine_learning  |                              │       ttl=settings.model_ttl, **kwargs))        │
immich_machine_learning  |                              │   117 │   model.configure(**kwargs)             │
immich_machine_learning  |                              │   118 │   outputs = await run(model.predict, in │
immich_machine_learning  |                              │   119 │   return ORJSONResponse(outputs)        │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │ /usr/src/app/main.py:137 in load                │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │   134 │   │   │   model.load()                  │
immich_machine_learning  |                              │   135 │                                         │
immich_machine_learning  |                              │   136 │   try:                                  │
immich_machine_learning  |                              │ ❱ 137 │   │   await run(_load, model)           │
immich_machine_learning  |                              │   138 │   │   return model                      │
immich_machine_learning  |                              │   139 │   except (OSError, InvalidProtobuf, Bad │
immich_machine_learning  |                              │   140 │   │   log.warning(                      │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │ /usr/src/app/main.py:125 in run                 │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │   122 async def run(func: Callable[..., Any], i │
immich_machine_learning  |                              │   123 │   if thread_pool is None:               │
immich_machine_learning  |                              │   124 │   │   return func(inputs)               │
immich_machine_learning  |                              │ ❱ 125 │   return await asyncio.get_running_loop │
immich_machine_learning  |                              │   126                                           │
immich_machine_learning  |                              │   127                                           │
immich_machine_learning  |                              │   128 async def load(model: InferenceModel) ->  │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │ /usr/local/lib/python3.11/concurrent/futures/th │
immich_machine_learning  |                              │ read.py:58 in run                               │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │ /usr/src/app/main.py:134 in _load               │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │   131 │                                         │
immich_machine_learning  |                              │   132 │   def _load(model: InferenceModel) -> N │
immich_machine_learning  |                              │   133 │   │   with lock:                        │
immich_machine_learning  |                              │ ❱ 134 │   │   │   model.load()                  │
immich_machine_learning  |                              │   135 │                                         │
immich_machine_learning  |                              │   136 │   try:                                  │
immich_machine_learning  |                              │   137 │   │   await run(_load, model)           │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │ /usr/src/app/models/base.py:52 in load          │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │    49 │   │   │   return                        │
immich_machine_learning  |                              │    50 │   │   self.download()                   │
immich_machine_learning  |                              │    51 │   │   log.info(f"Loading {self.model_ty │
immich_machine_learning  |                              │       to memory")                               │
immich_machine_learning  |                              │ ❱  52 │   │   self._load()                      │
immich_machine_learning  |                              │    53 │   │   self.loaded = True                │
immich_machine_learning  |                              │    54 │                                         │
immich_machine_learning  |                              │    55 │   def predict(self, inputs: Any, **mode │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │ /usr/src/app/models/clip.py:146 in _load        │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │   143 │   │   super().__init__(clean_name(model │
immich_machine_learning  |                              │   144 │                                         │
immich_machine_learning  |                              │   145 │   def _load(self) -> None:              │
immich_machine_learning  |                              │ ❱ 146 │   │   super()._load()                   │
immich_machine_learning  |                              │   147 │   │   self._load_tokenizer()            │
immich_machine_learning  |                              │   148 │   │                                     │
immich_machine_learning  |                              │   149 │   │   size: list[int] | int = self.prep │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │ /usr/src/app/models/clip.py:36 in _load         │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │    33 │   def _load(self) -> None:              │
immich_machine_learning  |                              │    34 │   │   if self.mode == "text" or self.mo │
immich_machine_learning  |                              │    35 │   │   │   log.debug(f"Loading clip text │
immich_machine_learning  |                              │ ❱  36 │   │   │   self.text_model = self._make_ │
immich_machine_learning  |                              │    37 │   │   │   log.debug(f"Loaded clip text  │
immich_machine_learning  |                              │    38 │   │                                     │
immich_machine_learning  |                              │    39 │   │   if self.mode == "vision" or self. │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │ /usr/src/app/models/base.py:117 in              │
immich_machine_learning  |                              │ _make_session                                   │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │   114 │   │   │   case ".armnn":                │
immich_machine_learning  |                              │   115 │   │   │   │   session = AnnSession(mode │
immich_machine_learning  |                              │   116 │   │   │   case ".onnx":                 │
immich_machine_learning  |                              │ ❱ 117 │   │   │   │   session = ort.InferenceSe │
immich_machine_learning  |                              │   118 │   │   │   │   │   model_path.as_posix() │
immich_machine_learning  |                              │   119 │   │   │   │   │   sess_options=self.ses │
immich_machine_learning  |                              │   120 │   │   │   │   │   providers=self.provid │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │ /opt/venv/lib/python3.11/site-packages/onnxrunt │
immich_machine_learning  |                              │ ime/capi/onnxruntime_inference_collection.py:43 │
immich_machine_learning  |                              │ 2 in __init__                                   │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │    429 │   │   │   │   │   self.disable_fallbac │
immich_machine_learning  |                              │    430 │   │   │   │   │   return               │
immich_machine_learning  |                              │    431 │   │   │   │   except Exception as fall │
immich_machine_learning  |                              │ ❱  432 │   │   │   │   │   raise fallback_error │
immich_machine_learning  |                              │    433 │   │   │   # Fallback is disabled. Rais │
immich_machine_learning  |                              │    434 │   │   │   raise e                      │
immich_machine_learning  |                              │    435                                          │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │ /opt/venv/lib/python3.11/site-packages/onnxrunt │
immich_machine_learning  |                              │ ime/capi/onnxruntime_inference_collection.py:42 │
immich_machine_learning  |                              │ 7 in __init__                                   │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │    424 │   │   │   │   │   print(f"EP Error {e} │
immich_machine_learning  |                              │    425 │   │   │   │   │   print(f"Falling back │
immich_machine_learning  |                              │    426 │   │   │   │   │   print("************* │
immich_machine_learning  |                              │ ❱  427 │   │   │   │   │   self._create_inferen │
immich_machine_learning  |                              │    428 │   │   │   │   │   # Fallback only once │
immich_machine_learning  |                              │    429 │   │   │   │   │   self.disable_fallbac │
immich_machine_learning  |                              │    430 │   │   │   │   │   return               │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │ /opt/venv/lib/python3.11/site-packages/onnxrunt │
immich_machine_learning  |                              │ ime/capi/onnxruntime_inference_collection.py:48 │
immich_machine_learning  |                              │ 3 in _create_inference_session                  │
immich_machine_learning  |                              │                                                 │
immich_machine_learning  |                              │    480 │   │   │   disabled_optimizers = set(di │
immich_machine_learning  |                              │    481 │   │                                    │
immich_machine_learning  |                              │    482 │   │   # initialize the C++ InferenceSe │
immich_machine_learning  |                              │ ❱  483 │   │   sess.initialize_session(provider │
immich_machine_learning  |                              │    484 │   │                                    │
immich_machine_learning  |                              │    485 │   │   self._sess = sess                │
immich_machine_learning  |                              │    486 │   │   self._sess_options = self._sess. │
immich_machine_learning  |                              ╰─────────────────────────────────────────────────╯
immich_machine_learning  |                              RuntimeError:                                      
immich_machine_learning  |                              /onnxruntime_src/onnxruntime/core/providers/cuda/cu
immich_machine_learning  |                              da_call.cc:121 std::conditional_t<THRW, void,      
immich_machine_learning  |                              onnxruntime::common::Status>                       
immich_machine_learning  |                              onnxruntime::CudaCall(ERRTYPE, const char*, const  
immich_machine_learning  |                              char*, ERRTYPE, const char*, const char*, int)     
immich_machine_learning  |                              [with ERRTYPE = cudaError; bool THRW = true;       
immich_machine_learning  |                              std::conditional_t<THRW, void,                     
immich_machine_learning  |                              onnxruntime::common::Status> = void]               
immich_machine_learning  |                              /onnxruntime_src/onnxruntime/core/providers/cuda/cu
immich_machine_learning  |                              da_call.cc:114 std::conditional_t<THRW, void,      
immich_machine_learning  |                              onnxruntime::common::Status>                       
immich_machine_learning  |                              onnxruntime::CudaCall(ERRTYPE, const char*, const  
immich_machine_learning  |                              char*, ERRTYPE, const char*, const char*, int)     
immich_machine_learning  |                              [with ERRTYPE = cudaError; bool THRW = true;       
immich_machine_learning  |                              std::conditional_t<THRW, void,                     
immich_machine_learning  |                              onnxruntime::common::Status> = void] CUDA failure  
immich_machine_learning  |                              100: no CUDA-capable device is detected ;          
immich_machine_learning  |                              GPU=-418058851 ; hostname=1736b8c4863e ;           
immich_machine_learning  |                              file=/onnxruntime_src/onnxruntime/core/providers/cu
immich_machine_learning  |                              da/cuda_execution_provider.cc ; line=245 ;         
immich_machine_learning  |                              expr=cudaSetDevice(info_.device_id);
```

### Additional information

nvidia-smi
Fri Jun 14 17:48:56 2024       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.171.04             Driver Version: 535.171.04   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  Quadro T2000                   On  | 00000000:01:00.0 Off |                  N/A |
| N/A   35C    P0               7W /  35W |      6MiB /  4096MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A      1117      G   /usr/lib/xorg/Xorg                            4MiB |
+---------------------------------------------------------------------------------------+

**Comment by bo0tzz** [comment]
This looks like https://github.com/NVIDIA/nvidia-container-toolkit/issues/48

**Comment by arnar-th** [comment]
> This looks like [NVIDIA/nvidia-container-toolkit#48](https://github.com/NVIDIA/nvidia-container-toolkit/issues/48)

I ran the steps under "How to check if you are affected" and it shows everything still working fine after deamon-reaload.

It got me thinking though and I checked my Jellyfin docker compose file and I have:
`    devices:
      - /dev/nvidia-caps:/dev/nvidia-caps
      - /dev/nvidia0:/dev/nvidia0
      - /dev/nvidiactl:/dev/nvidiactl
      - /dev/nvidia-modeset:/dev/nvidia-modeset
      - /dev/nvidia-uvm:/dev/nvidia-uvm
      - /dev/nvidia-uvm-tools:/dev/nvidia-uvm-tools
`
For some reason I don't remember (probably copy/pasted from somewhere).

Might try to 'mount' the device similarly in immich.

**Comment by arnar-th** [comment]
The
```
$ sudo systemctl daemon-reload
```
does not instantly cause nvidia/cuda to crash in the machine-learning container, and even after I run it nvidia-smi works fine on the host and other containers.

If I wait few minutes so that the ML model gets unloaded the container generates the error when it tries to load the model again:
```
immich_machine_learning  | [06/14/24 19:32:46] INFO     Starting gunicorn 22.0.0                           
immich_machine_learning  | [06/14/24 19:32:46] INFO     Listening at: http://[::]:3003 (9)                 
immich_machine_learning  | [06/14/24 19:32:46] INFO     Using worker: app.config.CustomUvicornWorker       
immich_machine_learning  | [06/14/24 19:32:46] INFO     Booting worker with pid: 21                        
immich_machine_learning  | [06/14/24 19:32:49] INFO     Started server process [21]                        
immich_machine_learning  | [06/14/24 19:32:49] INFO     Waiting for application startup.                   
immich_machine_learning  | [06/14/24 19:32:49] INFO     Created in-memory cache with unloading after 300s  
immich_machine_learning  |                              of inactivity.                                     
immich_machine_learning  | [06/14/24 19:32:49] INFO     Initialized request thread pool with 12 threads.   
immich_machine_learning  | [06/14/24 19:32:49] INFO     Application startup complete.                      
immich_machine_learning  | [06/14/24 19:32:51] INFO     Setting 'ViT-B-32__openai' execution providers to  
immich_machine_learning  |                              ['CUDAExecutionProvider', 'CPUExecutionProvider'], 
immich_machine_learning  |                              in descending order of preference                  
immich_machine_learning  | [06/14/24 19:32:51] INFO     Loading clip model 'ViT-B-32__openai' to memory    
immich_machine_learning  | 2024-06-14 19:32:52.369965564 [W:onnxruntime:, transformer_memcpy.cc:74 ApplyImpl] 15 Memcpy nodes are added to the graph main_graph for CUDAExecutionProvider. It might have negative impact on performance (including unable to run CUDA graph). Set session_options.log_severity_level=1 to see the detail logs before this message.
immich_machine_learning  | [06/14/24 19:39:29] INFO     Shutting down due to inactivity.                   
immich_machine_learning  | [06/14/24 19:39:29] INFO     Shutting down                                      
immich_machine_learning  | [06/14/24 19:39:29] INFO     Waiting for application shutdown.                  
immich_machine_learning  | [06/14/24 19:39:29] INFO     Application shutdown complete.                     
immich_machine_learning  | [06/14/24 19:39:29] INFO     Finished server process [21]                       
immich_machine_learning  | [06/14/24 19:39:29] ERROR    Worker (pid:21) was sent SIGINT!                   
immich_machine_learning  | [06/14/24 19:39:29] INFO     Booting worker with pid: 58                        
immich_machine_learning  | [06/14/24 19:39:32] INFO     Started server process [58]                        
immich_machine_learning  | [06/14/24 19:39:32] INFO     Waiting for application startup.                   
immich_machine_learning  | [06/14/24 19:39:32] INFO     Created in-memory cache with unloading after 300s  
immich_machine_learning  |                              of inactivity.                                     
immich_machine_learning  | [06/14/24 19:39:32] INFO     Initialized request thread pool with 12 threads.   
immich_machine_learning  | [06/14/24 19:39:32] INFO     Application startup complete.                      
immich_machine_learning  | [06/14/24 20:45:30] INFO     Setting 'ViT-B-32__openai' execution providers to  
immich_machine_learning  |                              ['CUDAExecutionProvider', 'CPUExecutionProvider'], 
immich_machine_learning  |                              in descending order of preference                  
immich_machine_learning  | [06/14/24 20:45:30] INFO     Loading clip model 'ViT-B-32__openai' to memory    
immich_machine_learning  | *************** EP Error ***************
immich_machine_learning  | EP Error /onnxruntime_src/onnxruntime/core/providers/cuda/cuda_call.cc:121 std::conditional_t<THRW, void, onnxruntime::common::Status> onnxruntime::CudaCall(ERRTYPE, const char*, const char*, ERRTYPE, const char*, const char*, int) [with ERRTYPE = cudaError; bool THRW = true; std::conditional_t<THRW, void, onnxruntime::common::Status> = void] /onnxruntime_src/onnxruntime/core/providers/cuda/cuda_call.cc:114 std::conditional_t<THRW, void, onnxruntime::common::Status> onnxruntime::CudaCall(ERRTYPE, const char*, const char*, ERRTYPE, const char*, const char*, int) [with ERRTYPE = cudaError; bool THRW = true; std::conditional_t<THRW, void, onnxruntime::common::Status> = void] CUDA failure 100: no CUDA-capable device is detected ; GPU=32490 ; hostname=3eadd8d5a82d ; file=/onnxruntime_src/onnxruntime/core/providers/cuda/cuda_execution_provider.cc ; line=245 ; expr=cudaSetDevice(info_.device_id); 
immich_machine_learning  | 
immich_machine_learning  |  when using ['CUDAExecutionProvider', 'CPUExecutionProvider']
immich_machine_learning  | Falling back to ['CUDAExecutionProvider', 'CPUExecutionProvider'] and retrying.
```

**Comment by arnar-th** [comment]
Scrap that.

Can recreate the issue with:
```
~/docker/immich$ docker exec -it immich_machine_learning nvidia-smi
Fri Jun 14 21:20:13 2024       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.171.04             Driver Version: 535.171.04   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  Quadro T2000                   On  | 00000000:01:00.0 Off |                  N/A |
| N/A   35C    P0               7W /  35W |      6MiB /  4096MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
+---------------------------------------------------------------------------------------+
~/docker/immich$ sudo systemctl daemon-reload
~/docker/immich$ docker exec -it immich_machine_learning nvidia-smi
Failed to initialize NVML: Unknown Error
```

Probably has nothing to do with immich so I'm closing this issue.  Thanks for the help @bo0tzz !
